swagger.yaml

```diff
@@ -1,75 +1,208 @@
+          schema:
+            $ref: "#/definitions/Error"
+
+  /textbooks:
+    get:
+      tags:
+      - "textbooks"
+      summary: "Retrieve textbooks by parameters"
+      operationId: "getTextbooks"
+      parameters:
+
+        - name: "term"
+          in: "query"
+          description: "Name of term in which to find textbooks"
+          type: "string"
+          required: true
+
+        - name: "department"
+          in: "query"
+          description: "Name of department in which to find textbooks"
+          type: "string"
+          required: true
 
-parameters:
-  pretty:
-    name: pretty
-    in: query
-    type: boolean
-    required: false
-    description: If true, JSON response will be pretty-printed
+        - name: "course"
+          in: "query"
+          description: "Name of course in which to find textbooks"
+          type: "string"
+          required: true
+
+        - name: "section"
+          in: "query"
+          description: "Name of section in which to find textbooks"
+          type: "string"
+          required: false
+
+        - name: "isRequired"
+          in: "query"
+          description: "If true, only retrieves required textbooks"
+          type: "string"
+          required: false
+          default: false
+
+      responses:
+        200:
+          description: "Textbooks successfully retrieved"
+          schema:
+            $ref: "#/definitions/TextbookResultArray"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
 
 definitions:
-  Info:
+
+  Textbook:
+    type: "object"
     properties:
-      name:
-        type: string
-        description: Name of the API.
-      time:
-        type: string
-        description: Build time in ISO 8601 format.
-      unixTime:
-        type: integer
-        description: Build time as milliseconds since the unix epoch.
-      commit:
-        type: string
-        description: Git commit hash of the revision which was built.
-      documentation:
-        type: string
-        description: Filename of the swagger specification for the API.
+      isbn:
+        type: "string"
+        description: "ISBN of textbook"
+        example: "9780998819112"
+      required:
+        type: "string"
+        description: "Required status of textbook"
+        example: "Required"
+      coverImageUrl:
+        type: "string"
+        description: "URL of cover image"
+        example: "coverimages.verbacompete.com/no_image.jpg"
+      title:
+        type: "string"
+        description: "Title of textbook"
+        example: "Probability & Stat For Engr & Sci"
+      author:
+        type: "string"
+        description: "Author of textbook"
+        example: "Devore"
+      edition:
+        type: "string"
+        description: "Textbook edition"
+        example: "9"
```

Comment:
Same here, could this be an integer? If the data you're seeing is unpredictable, then don't worry about it. For example, I'm not sure if the edition could ever contain characters.

Reply:
I made some of the variables (price, edition, copyright_year) strings just to be conservative since I wasn't sure if there were cases where a book had some strange value for these fields. It should be fine to change these though. For items listed as textbooks that aren't really books (turing clickers), edition and copyright_year are just null.